repair_type/models/stock_move.py

```diff
@@ -35,3 +35,16 @@ def _get_repair_locations(self, repair_line_type, repair_id=False):
                     res[1],
                 )
         return res
+
+    def _compute_location_id(self):
+        res = True
+        for move in self:
+            if move.repair_line_type and move.repair_id:
+                location_src, location_dest = move._get_repair_locations(
+                    move.repair_line_type
+                )
+                move.location_id = location_src
+                move.location_dest_id = location_dest
+            else:
+                res = super()._compute_location_id()
```

Comment:
I think this is also not correct, you are doing super for the whole recordset on every iteration. The odoo approach can inspire you https://github.com/odoo/odoo/commit/6aee84ea605c600f2b2d72251d8517c985ab150f#diff-ff31dd528a40b4d1e4dffeabce24c78b96deb6576802956f532eaaeb9826cbd8R257